cypress/integration/creatorOnboardingFlows/creatorSignup.spec.js

```diff
@@ -0,0 +1,100 @@
+  });
+
+  it('should display instructions', () => {
+    cy.findByText('Create your admin account first.').should('be.visible');
+    cy.findByText("Then we'll walk you through your Forem setup.").should(
+      'be.visible',
+    );
+  });
+
+  it('should display a validated username hint that correlates to the name entered', () => {
+    cy.findByTestId('creator-signup-form').as('creatorSignupForm');
+    cy.get('@creatorSignupForm')
+      .findByText(/^Name$/)
+      .type('1 Forem creator name! Also test a maximum length string');
+
+    // restricts the string to 20 characters, lowercased,
+    // replaces spaces and special_characters with an underscore
+    cy.contains('1_forem_creator_name__also_tes');
+  });
+
+  it('should show and focus on the username field when clicking on the edit icon', () => {
+    cy.findByTestId('creator-signup-form').as('creatorSignupForm');
+    cy.get('@creatorSignupForm')
+      .findByText(/^Name$/)
+      .type('Forem creator name');
+
+    cy.findByRole('button', { name: 'Edit username' }).click();
+    cy.get('@creatorSignupForm')
+      .findByText(/^Username/)
+      .should('exist');
+
+    cy.get('input[name="user[username]"]').should(
+      'have.value',
+      'forem_creator_name',
+    );
+  });
+
+  it('should contain an email label and field', () => {
+    cy.findByTestId('creator-signup-form').as('creatorSignupForm');
+    cy.get('@creatorSignupForm')
+      .findByText(/^Email/)
+      .should('be.visible');
+  });
+
+  it('should contain an password', () => {
+    cy.findByTestId('creator-signup-form').as('creatorSignupForm');
+    cy.get('@creatorSignupForm')
+      .findByText(/^Password/)
+      .should('be.visible');
+  });
+
+  it('should toggle the password when the eye icon is clicked', () => {
+    cy.findByTestId('creator-signup-form').as('creatorSignupForm');
+    cy.get('@creatorSignupForm')
+      .findByText(/^Password$/)
+      .type('abc123456');
+
+    cy.get('input[name="user[password]"]').should(
+      'have.attr',
+      'type',
+      'password',
+    );
+    cy.get('.js-eye').should('be.visible');
+    cy.get('.js-eye-off').should('not.be.visible');
+
+    cy.findByRole('button', { name: 'Toggle password visibility' }).click();
+    cy.get('input[name="user[password]"]').should('have.attr', 'type', 'text');
+    cy.get('.js-eye-off').should('be.visible');
+    cy.get('.js-eye').should('not.be.visible');
+  });
+
+  it("should allow sign the user in when 'Create my account' is clicked", () => {
+    cy.findByTestId('creator-signup-form').as('creatorSignupForm');
+    cy.get('@creatorSignupForm').findByText(/^Name/).type('Forem Creator');
+
+    cy.get('@creatorSignupForm')
+      .findByText(/^Email/)
+      .type('forem_creator@gmail.com');
+
+    cy.get('@creatorSignupForm')
+      .findByText(/^Password/)
+      .type('abc123456');
+
+    cy.get('@creatorSignupForm')
+      .findAllByRole('button', { name: 'Create my account' })
+      .last()
+      .click();
```

Comment:
Refactor (blocking): There's only one *Create my account* button. Use `.findByRole` instead.

```suggestion
      .findByRole('button', { name: 'Create my account' })
      .click();
```

Reply:
In regards to the test failing on CI, even though it clicks the _Create my account_ button, it's not actually doing anything from what I can tell.

The refactor above won't fix that, but I wonder if an application error I saw in the logs might be the culprit. It doesn't show up all the time, but locally I saw an error about being unable to access `.classList` on `null`, which means there is a DOM element reference in the application code potentially causing the issue. It probably runs fast enough locally most of the time which is why yo potentially don't get a failure locally. Just a hunch.